src/Razor/Microsoft.AspNetCore.Razor.Language/test/StringSegmentTest.cs

```diff
@@ -568,6 +568,60 @@ public void IndexOf_SearchOnlyInsideTheRange_IfStartAndCountAreProvided()
             Assert.Equal(-1, result);
         }
 
+        [Fact]
```

Comment:
Nit: let's add a note in `StringSegment.cs` to explain why we're rolling out our own implementation.

Reply:
Added a doc comment to the type.